src/util/types/typeChecker.ts

```
@@ -977,7 +976,7 @@ export function createTypeChecker(workspace: IElmWorkspace): TypeChecker {
     ).diagnostics.forEach((diagnostic) => diagnostics.add(diagnostic));
   }
 
-  function getSourceFileOfNode(node: SyntaxNode): ITreeContainer {
-    return forest.getByUri(node.tree.uri)!;
+  function getSourceFileOfNode(node: SyntaxNode): ITreeContainer | undefined {
```

If you prefer this, that's fine. But it should always be defined, and it adds unnecessary checking since it returns undefined. 

Maybe we could just throw an error if not defined, and then we could just return `ITreeContainer`?

---

Changed